### PR TITLE
Add enable_mask argument on vm list

### PR DESCRIPTION
**Pull request by marvcoa**
enable masking of subscription ids for privacy

**Description**

az vm --list 명령어를 사용할때 굳이 알려줄필요가 없는 개인정보가 다수 포함되어있는데 이런 개인정보를 마스킹 할수있는 명령어가 필요해  params와 custum파일을 수정하여 명령어를 추가하였음

**Testing Guide**

az vm list --enablemask or az vm list -m을 입력하면 실행이 가능함

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -1417,6 +1421,9 @@ def show_vm(cmd, resource_group_name, vm_name, show_details=False, include_user_
     if show_details:
         return get_vm_details(cmd, resource_group_name, vm_name, include_user_data)
 
+    if mask:
```
Comment (ianychoi) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py]: Plz remove L1424-26

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 2]
```diff
@@ -788,7 +788,7 @@ def create_vm(cmd, vm_name, resource_group_name, image=None, size='Standard_DS1_
               enable_hotpatching=None, platform_fault_domain=None, security_type=None, enable_secure_boot=None,
               enable_vtpm=None, count=None, edge_zone=None, nic_delete_option=None, os_disk_delete_option=None,
               data_disk_delete_option=None, user_data=None, capacity_reservation_group=None, enable_hibernation=None,
-              v_cpus_available=None, v_cpus_per_core=None, accept_term=None):
+              v_cpus_available=None, v_cpus_per_core=None, accept_term=None,mask=None):
```
Comment (ianychoi) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py]: enable_mask?

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 3]
```diff
@@ -517,6 +519,10 @@ def load_arguments(self, _):
         c.argument('image_location', get_location_type(self.cli_ctx))
         c.argument('edge_zone', edge_zone_type)
 
+    with self.argument_context('mask') as c:
```
Comment (ianychoi) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py]: Does this part required for this change?

**Review by ianychoi — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 3 above)
Comment (marvcoa) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: No so deleted this part

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 2 above)
Comment (marvcoa) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: yes thanks!

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (marvcoa) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: Thank you - done!

**Pull request merged**